Sources/W3CTraceContext/SpanID.swift

```diff
@@ -51,66 +55,115 @@ public struct SpanID: Sendable {
+            try Swift.withUnsafeBytes(of: _bytes) { bytes in
+                try bytes.withMemoryRebound(to: UInt8.self, body)
+            }
+        }
+
+        @inlinable
+        public func withUnsafeBytes<Result>(_ body: (UnsafeRawBufferPointer) throws -> Result) rethrows -> Result {
+            try Swift.withUnsafeBytes(of: _bytes, body)
+        }
+
+        @inlinable
+        public mutating func withUnsafeMutableBytes<R>(_ body: (UnsafeMutableRawBufferPointer) throws -> R) rethrows -> R {
+            try Swift.withUnsafeMutableBytes(of: &_bytes) { bytes in
+                try body(bytes)
+            }
+        }
+
+        public static func == (lhs: Self, rhs: Self) -> Bool {
+            lhs._bytes.0 == rhs._bytes.0
+                && lhs._bytes.1 == rhs._bytes.1
+                && lhs._bytes.2 == rhs._bytes.2
+                && lhs._bytes.3 == rhs._bytes.3
+                && lhs._bytes.4 == rhs._bytes.4
+                && lhs._bytes.5 == rhs._bytes.5
+                && lhs._bytes.6 == rhs._bytes.6
+                && lhs._bytes.7 == rhs._bytes.7
+        }
+
+        public func hash(into hasher: inout Hasher) {
+            hasher.combine(_bytes.0)
+            hasher.combine(_bytes.1)
+            hasher.combine(_bytes.2)
+            hasher.combine(_bytes.3)
+            hasher.combine(_bytes.4)
+            hasher.combine(_bytes.5)
+            hasher.combine(_bytes.6)
+            hasher.combine(_bytes.7)
+        }
     }
 }
 
-extension SpanID: Hashable {
-    public func hash(into hasher: inout Hasher) {
-        hasher.combine(_bytes.0)
-        hasher.combine(_bytes.1)
-        hasher.combine(_bytes.2)
-        hasher.combine(_bytes.3)
-        hasher.combine(_bytes.4)
-        hasher.combine(_bytes.5)
-        hasher.combine(_bytes.6)
-        hasher.combine(_bytes.7)
+extension SpanID.Bytes: CustomStringConvertible {
+    /// A 16 character hex string representation of the bytes.
+    public var description: String {
+        String(decoding: hexBytes, as: UTF8.self)
+    }
+
+    /// A 16 character UTF-8 hex byte array representation of the bytes.
+    public var hexBytes: [UInt8] {
+        var asciiBytes = [UInt8](repeating: 0, count: 16)
+        for i in startIndex ..< endIndex {
+            let byte = self[i]
+            asciiBytes[2 * i] = Hex.lookup[Int(byte >> 4)]
+            asciiBytes[2 * i + 1] = Hex.lookup[Int(byte & 0x0F)]
+        }
+        return asciiBytes
     }
 }
 
+extension SpanID: Equatable {}
+
+extension SpanID: Hashable {}
+
 extension SpanID: Identifiable {
-    public var id: [UInt8] { bytes }
+    public var id: Self { self }
 }
 
 extension SpanID: CustomStringConvertible {
     /// A 16 character hex string representation of the span ID.
     public var description: String {
-        String(decoding: hexBytes, as: UTF8.self)
+        String(decoding: bytes.hexBytes, as: UTF8.self)
```

Comment:
```suggestion
        String(describing: bytes)
```
How about making use of `SpanID.Bytes`'s conformance to `CustomStringConvertible`?